uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -246,10 +273,13 @@ define([
             return $q.all[initUserRole(), initOpenMode()];
           })
           .then(function () {
-            return AbsencePeriod.current();
+            return loadAbsencePeriods();
           })
-          .then(function (absencePeriodInstance) {
-            vm.period = absencePeriodInstance;
+          .then(function () {
```

Comment:
I think sequence of then's keep the code clean and related code is kept in one block